File: pyg_cart_web/src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService',function($http){
	    	

	//获取购物车信息
	this.findCartList=function () {
		return $http.get('../cart/findCartList.do?');
    }
    //获取用户名
    this.findName=function () {
        return $http.get('../cart/findName.do?');
    }
    //添加商品到购物车
	this.addGoodsToCartList=function (itemId,num) {
        return $http.get('../cart/addGoodsToCartList.do?itemId='+itemId+"&num="+num);
    }
});

File: pyg_cart_interface/src/main/java/com/pyg/cart/service/CartService.java
package com.pyg.cart.service;

import entity.Cart;

import java.util.List;

public interface CartService {
    /**
     * 添加商品
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

    /**
     * 从redis中查询购物车
     * @return
     */
    List<Cart> findCartListFromRedis(String username);
    /**
     * 将购物车保存到redis
     */
    void saveCartListToRedis(String username,List<Cart> cartList);

    /**
     * 合并购物车
     * @param cartList1
     * @param cartList2
     * @return
     */
    List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);

}
